File: speech_to_text.py
import io
from google.oauth2 import service_account
from google.cloud import speech
import dotenv
dotenv.load_dotenv()

client = speech.SpeechClient()

fileName = "What is Fortnite's Best Shotgun_.mp3";
with io.open(fileName, 'rb') as f:
    content = f.read()
    audio = speech.RecognitionAudio(content=content)

config = speech.RecognitionConfig(
    encoding=speech.RecognitionConfig.AudioEncoding.MP3,
    sample_rate_hertz=16000,
    language_code='en-US'
)

response = client.recognize(config=config, audio=audio)

print(response)

File: speech_to_text.js
const speech = require("@google-cloud/speech");
const fs = require("fs");
const dotenv = require("dotenv");
dotenv.config();

async function main() {
  const client = new speech.SpeechClient();
  const fileName = "./s-to-t_test.m4a";

  const file = fs.readFileSync(fileName);
  const audioBytes = file.toString("base64");

  const audio = {
    content: audioBytes,
  };

  const config = {
    encoding: "LINEAR16",
    sampleRateHertz: 16000,
    languageCode: "en-US",
  };
  const request = {
    audio: audio,
    config: config,
  };

  const [response] = await client.recognize(request);
  const transcription = response.results
    .map((result) => result.alternatives[0].transcript)
    .join("\n");
  console.log(`Transcript: ${transcription}`);
}

main().catch(console.error);
